api.py

```diff
@@ -37,24 +86,26 @@ def get(self):
 		selected_semester = self.request.get('semester')
 		
 		if selected_semester:
-			members = Member.query(Member.show == True, Member.semesters_paid == selected_semester).order(Member.name).fetch(limit=None)
+			# Avoid throwing error 500 if a bad semester string is supplied
+			try:
+				selected_semester = float(selected_semester)
+			except ValueError:
+				self.error(400)
+				return
+			members = Member.query(Member.semesters_paid == selected_semester).order(Member.name).fetch(limit=None)
 		else:
-			members = Member.query(Member.show == True, Member.never_paid == False).order(Member.name).fetch(limit=None)
+			members = Member.query().order(Member.name).fetch(limit=None)
```

Comment:
This appears to expose members who asked to not appear on the website (the purpose of the `Member.show == True`), as well as non-members in the system beacuse they ran a mission or are on the mailing list (`Member.never_paid == False`).

Reply:
I thought that it would be simpler to provide access to all members through the API, and do the member filtering in the app that pulls data, since there could be cases where we might want access to data for members that have not paid or don't wish to be seen publicly. Since access to the API is gated by the API key anyway, better to give more information here and avoid limiting uses of this interface later on down the road.

Reply:
I am pretty sure the database has more non-members than members (between the mailing list and non-members who ran missions).  Fetching every member for every API call is going to make a lot of queries unnecessarily slow and put a lot of unnecessary load on the server.

What do you think about adding a URL parameter for including non-members in the system that is `False` by default?

Reply:
Sure, makes sense. I will implement a default `False` parameter to include private or non-paid members, that way  we can only pull the whole list when necessary